tests/execution_engine/split_and_sample/test_sqlalchemy_execution_engine_sampling.py

```diff
@@ -0,0 +1,197 @@
+            GESqlDialect.SQLITE: "SELECT * FROM TEST_SCHEMA_NAME.TEST_TABLE WHERE 1 = 1 LIMIT 10 OFFSET 0",
+            GESqlDialect.BIGQUERY: "SELECT * FROM `TEST_SCHEMA_NAME`.`TEST_TABLE` WHERE TRUE LIMIT 10",
+            GESqlDialect.SNOWFLAKE: "SELECT * FROM TEST_SCHEMA_NAME.TEST_TABLE WHERE TRUE LIMIT 10",
+            GESqlDialect.REDSHIFT: "SELECT * FROM TEST_SCHEMA_NAME.TEST_TABLE WHERE TRUE LIMIT 10",
+            GESqlDialect.AWSATHENA: 'SELECT * FROM "TEST_SCHEMA_NAME"."TEST_TABLE" WHERE TRUE LIMIT 10',
+            GESqlDialect.DREMIO: 'SELECT * FROM "TEST_SCHEMA_NAME"."TEST_TABLE" WHERE 1 = 1 LIMIT 10',
+            GESqlDialect.TERADATASQL: "SELECT TOP 10 * FROM TEST_SCHEMA_NAME.TEST_TABLE WHERE 1 = 1",
+        }
+        return DIALECT_NAME_TO_SQL_STATEMENT[dialect_name]
+
+    return _dialect_name_to_sql_statement
+
+
+@pytest.mark.parametrize(
+    "dialect_name",
+    [
+        pytest.param(dialect_name, id=dialect_name.value)
+        for dialect_name in GESqlDialect.get_all_dialects()
+    ],
+)
+def test_sample_using_limit_builds_correct_query_where_clause_none(
+    dialect_name: GESqlDialect, dialect_name_to_sql_statement, sa
+):
+    """What does this test and why?
+
+    split_on_limit should build the appropriate query based on input parameters.
+    This tests dialects that differ from the standard dialect, not each dialect exhaustively.
+    """
+
+    # 1. Setup
+    class MockSqlAlchemyExecutionEngine:
+        def __init__(self, dialect_name: GESqlDialect):
+            self._dialect_name = dialect_name
+            self._connection_string = self.dialect_name_to_connection_string(
+                dialect_name
+            )
+
+        DIALECT_TO_CONNECTION_STRING_STUB: dict = {
+            GESqlDialect.POSTGRESQL: "postgresql://",
+            GESqlDialect.MYSQL: "mysql+pymysql://",
+            GESqlDialect.ORACLE: "oracle+cx_oracle://",
+            GESqlDialect.MSSQL: "mssql+pyodbc://",
+            GESqlDialect.SQLITE: "sqlite:///",
+            GESqlDialect.BIGQUERY: "bigquery://",
+            GESqlDialect.SNOWFLAKE: "snowflake://",
+            GESqlDialect.REDSHIFT: "redshift+psycopg2://",
+            GESqlDialect.AWSATHENA: f"awsathena+rest://@athena.us-east-1.amazonaws.com/some_test_db?s3_staging_dir=s3://some-s3-path/",
+            GESqlDialect.DREMIO: "dremio://",
+            GESqlDialect.TERADATASQL: "teradatasql://",
+        }
+
+        @property
+        def dialect_name(self) -> str:
+            return self._dialect_name.value
+
+        def dialect_name_to_connection_string(self, dialect_name: GESqlDialect) -> str:
+            return self.DIALECT_TO_CONNECTION_STRING_STUB.get(dialect_name)
+
+        _BIGQUERY_MODULE_NAME = "sqlalchemy_bigquery"
+
+        @property
+        def dialect(self) -> sa.engine.Dialect:
+            # TODO: AJB 20220512 move this dialect retrieval to a separate class from the SqlAlchemyExecutionEngine
+            #  and then use it here.
```

Comment:
For all these todos, are they tracked in Jira in any way? If not, is that something we should do to prevent losing track of the item?

Reply:
No :/ I'm hoping to get this one in soon though before I lose context of this part of the codebase.

Reply:
FYI this TODO is now created as https://superconductive.atlassian.net/browse/GREAT-893 for tracking.